contracts/market-wrapper/NounsMarketWrapper.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports ============
+import {INounsAuctionHouse} from "../external/interfaces/INounsAuctionHouse.sol";
+
+// ============ Internal Imports ============
+import {IMarketWrapper} from "./IMarketWrapper.sol";
+
+/**
+ * @title NounsMarketWrapper
+ * @author Anna Carroll + Nounders
+ * @notice MarketWrapper contract implementing IMarketWrapper interface
+ * according to the logic of Nouns' Auction Houses
+ */
+contract NounsMarketWrapper is IMarketWrapper {
+    // ============ Public Immutables ============
+
+    INounsAuctionHouse public immutable market;
+
+    // ======== Constructor =========
+
+    constructor(address _nounsAuctionHouse) {
+        market = INounsAuctionHouse(_nounsAuctionHouse);
+    }
+
+    // ======== External Functions =========
+
+    /**
+     * @notice Determine whether there is an existing, active auction
+     * for this token. In the Nouns auction house, the current auction
+     * id is the token id, which increments sequentially, forever. The
+     * auction is considered active while the current block timestamp
+     * is less than the auction's end time.
+     * @return TRUE if the auction exists
+     */
+    function auctionExists(uint256 auctionId)
+      public
+      view
+      override
+      returns (bool)
+    {
+        (uint256 currentAuctionId, , , uint256 endTime, , ) = market.auction();
+        return auctionId == currentAuctionId && block.timestamp < endTime;
+    }
+
+    /**
+     * @notice Determine whether the given auctionId and tokenId is active.
+     * We ignore nftContract since it is static for all nouns auctions.
+     * @return TRUE if the auctionId and tokenId matches the active auction
+     */
+    function auctionIdMatchesToken(
+        uint256 auctionId,
+        address /* nftContract */,
+        uint256 tokenId
+    ) public view override returns (bool) {
+        return auctionId == tokenId && auctionExists(auctionId);
+    }
+
+    /**
+     * @notice Calculate the minimum next bid for the active auction
+     * @return minimum bid amount
+     */
+    function getMinimumBid(uint256 auctionId)
+      external
+      view
+      override
+      returns (uint256)
+    {
+        require(
+            auctionExists(auctionId),
+            "NounsMarketWrapper::getMinimumBid: Auction does not exist"
```

Comment:
nit: could say "Auction not active" here and line 97

Reply:
(addressed this but can't push)

Reply:
https://github.com/PartyDAO/partybid/pull/43/commits/7fd8dfc80ecfd3aca8bc6d2472d75f51eb59fcfa